lib/rules/no-decorators-before-export.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/** @type {import('eslint').Rule.RuleModule} */
+
+function reportInvalidSyntax(context, node) {
+  if (isInvalidSyntax(node)) {
+    const isDefaultExport = node.type === 'ExportDefaultDeclaration';
+    const isNamed = node.declaration.id !== null;
+    const name = isNamed ? node.declaration.id.name : '';
+    const numDecorators = node.declaration.decorators.length;
+
+    const message = `Usage of class decorator${numDecorators > 1 ? 's' : ''} on the ${
+      isNamed ? '' : 'un-named '
+    }${isDefaultExport ? 'default ' : ''}export${
+      isNamed ? ` ${name}` : ''
+    } must occur prior to exporting the class.`;
+
+    if (isNamed) {
+      const sourceCode = context.getSourceCode();
+      const src = sourceCode.getText(node);
+      context.report({
+        node,
+        message,
+        fix(fixer) {
+          if (isDefaultExport) {
+            let newSrc = src.replace(`export default class ${name}`, `class ${name}`);
+            newSrc = `${newSrc}\nexport default ${name};\n`;
+            return fixer.replaceText(node, newSrc);
+          }
+          let newSrc = src.replace(`export class ${name}`, `class ${name}`);
+          newSrc = `${newSrc}\nexport { ${name} };\n`;
+          return fixer.replaceText(node, newSrc);
+        },
+      });
+    } else {
+      context.report({ node, message });
+    }
+  }
+}
+
+function isInvalidSyntax(node) {
```

Comment:
Seems like this could just be a generic `classDeclarationHasDecorators(node)` function. That way, it's more clear what it's doing, and could be reusable in other places.

Reply:
It isn’t though, it’s checking against an export declaration specifically

Reply:
My bad, then `exportDeclarationHasDecorators`?. Feel free to leave it if you want. I'm just thinking of ways to make it more clear and reusable what it's doing.